horizon/static/framework/conf/resource-type-registry.service.js

```diff
@@ -477,13 +483,16 @@
      * If a configuration is supplied, the resource type is extended to
      * use the configuration's properties.
      */
-    function getResourceType(type, config) {
+    function getResourceType(type, config, slug) {
```

Comment:
The problem is that it's not setting anything on the type, it's on the registry itself, so it's just awkward because you have to have a separate line/call to register it.